### PR TITLE
Group unknown routes

### PR DESCRIPTION
**Pull request by kanadaj**
Creates a grouping for "Unknown Route" instead of using the path name for 404 responses.

Also exposes an `out` bool on `GetTransactionName` to make this simpler.

_#skip-changelog_

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]
LGTM. @Tyrrrz ?

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs [hunk 1]
```diff
@@ -44,6 +44,8 @@ public async Task InvokeAsync(HttpContext context)
             }
             finally
             {
+                transaction.Name = context.GetTransactionName();
```
Comment (Tyrrrz) [inline src/Sentry.AspNetCore/SentryTracingMiddleware.cs:47]: Let's add a comment here to explain why we get the transaction name again

**Review by Tyrrrz — APPROVED** [review]

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs (hunk 1 above)
Comment (kanadaj) [reply src/Sentry.AspNetCore/SentryTracingMiddleware.cs:47]: @Tyrrrz Not "again", if you check a few lines above I've replaced the default value with "Unknown Route"

**Comment by Tyrrrz** [comment]
#skip-changelog

**Pull request merged**